tests/ludwig/automl/test_data_source.py

```diff
@@ -22,15 +22,23 @@
 
 @pytest.mark.distributed
 def test_mixed_csv_data_source(ray_cluster_2cpu):
-    try:
-        temp = tempfile.NamedTemporaryFile(mode="w+")
-        temp.write(CSV_CONTENT)
-        temp.seek(0)
-        ds = read_csv(temp.name, dtype=None)
-        df = dd.from_pandas(ds, npartitions=1)
-        config = create_auto_config(dataset=df, target=[], time_limit_s=3600, tune_for_memory=False)
-        assert len(config["input_features"]) == 2
-        assert config["input_features"][0]["type"] == TEXT
-        assert config["input_features"][1]["type"] == TEXT
-    finally:
-        temp.close()
+    temp = tempfile.NamedTemporaryFile(mode="w+")
+    temp.write(CSV_CONTENT)
+    temp.seek(0)
+    ds = read_csv(temp.name, dtype=None)
+    df = dd.from_pandas(ds, npartitions=1)
+    config = create_auto_config(dataset=df, target=[], time_limit_s=3600, tune_for_memory=False)
+    assert len(config["input_features"]) == 2
+    assert config["input_features"][0]["type"] == TEXT
+    assert config["input_features"][1]["type"] == TEXT
+
+
+@pytest.mark.distributed
+def test_mixed_csv_data_source_with_profile(ray_cluster_2cpu):
+    temp = tempfile.NamedTemporaryFile(mode="w+")
+    temp.write(CSV_CONTENT)
+    temp.seek(0)
+    ds = read_csv(temp.name, dtype=None)
+    df = dd.from_pandas(ds, npartitions=1)
```

Comment:
Can we create a common shared utility function for these to be used by both `test_mixed_csv_data_source` and `test_mixed_csv_data_source_with_profile`?

Reply:
Done.